docs/fml-spec.md

```diff
@@ -0,0 +1,576 @@
+# The Feature Manifest Language
+
+## About this document
+
+This document is the specfiication for the Feature Manifest Language for use with the Nimbus SDK.
+
+Language features that have not yet been implemented are marked as so.
+
+## Introduction
+
+Nimbus is an experimentation platform to allow product owners to easily run experiments on their mobile apps.
+
+The [Feature API][feature-api] allows application developers to query Nimbus for JSON objects which they could use to configure the app's features.
+
+The Feature Manifest is a significant evolution of the Feature API: by generating Kotlin and Swift code, it adds type-safety, schema checking and defaulting to the Feature API.
+
+## High level concepts
+
+To the product owner, Nimbus allows you to experiment with different configurations of the application.
+
+It does this by exposing a small proportion of the audience to different configurations of a feature. Once a winning configuration is identified, it can be rolled out to the rest of the audience, all without a re-release of the app.
+
+To the app developer, **Nimbus is a configuration store**. Features are configured using data from Nimbus. Nimbus is updated at startup of each run.
```

Comment:
> Nimbus is updated at startup of each run.

This statement is hiding a bit of complexity here. Would it be worth linking to existing documentation or clarifying the difference between "updating" and "applying" here?

Reply:
I'm trying to keep the scope of this document to "the FML language spec". I deliberately don't go into much detail apart from that.

e.g. installing, running configuring the FML.